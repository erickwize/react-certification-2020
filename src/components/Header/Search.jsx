import React from "react";
import styled from "styled-components";

const Input = styled.input`
  font-size: 16px;
  color: #c0c0c0;
  background-color: #202020;
  border-radius: 3px;
  margin: 0 auto;
  width: 300px;
  padding: 8px;
`;

const Search = () => {
  return <Input placeholder="Search"></Input>;
};

export default Search;
