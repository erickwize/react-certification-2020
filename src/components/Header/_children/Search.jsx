import React, { useState } from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const Form = styled.form`
  display: flex;
  align-items: center;
  margin: 0 auto;
  width: 300px;

  #text {
    font-size: 16px;
    color: #c0c0c0;
    background-color: #202020;
    border-radius: 3px 0 0 3px;
    width: 85%;
    padding: 8px;
  }

  #submit {
    font-size: 20px;
    color: #a0a0a0;
    background-color: #202020;
    border-left: 1px solid #101010;
    border-radius: 0 3px 3px 0;
    width: 15%;
    cursor: pointer;
  }

  #submit:hover {
    background-color: #181818;
  }

  #submit:active {
    color: #d0d0d0;
  }
`;

const Search = ({ handleSearch }) => {
  const [text, setText] = useState("");

  const onChange = (e) => {
    setText(e.target.value);
  };
  const onClick = () => {
    document.getElementById("text").select();
  };

  const onSubmit = (e) => {
    e.preventDefault();
    if (text !== "") {
      handleSearch(text);
      document.getElementById("text").select();
    } else {
      console.log("Enter a search text");
    }
  };

  return (
    <Form onSubmit={onSubmit}>
      <input
        id="text"
        type="text"
        name="text"
        placeholder="Search"
        value={text}
        onChange={onChange}
        onClick={onClick}
      />
      <input
        id="submit"
        type="submit"
        className="material-icons"
        value="search"
      />
    </Form>
  );
};

Search.propTypes = {
  handleSearch: PropTypes.func.isRequired,
};

export default Search;
