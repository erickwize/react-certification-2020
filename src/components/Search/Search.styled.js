import styled from 'styled-components'

const StyledSearch = styled.div.attrs()`
  padding: 0.1em;
`
export default StyledSearch
