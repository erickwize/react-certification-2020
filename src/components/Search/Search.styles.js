import styled from 'styled-components';
import { SearchAlt } from '@styled-icons/boxicons-regular/SearchAlt';

export const InputContainer = styled.form`
  background-color: #3a7da7;
  padding: 3px;
  border-radius: 3px;
  margin-left: 16px;
`;

export const Icon = styled(SearchAlt)`
  color: #fff;
  width: 24px;
  height: 24px;
  margin-left: 5px;
`;

export const Input = styled.input`
  background-color: transparent;
  border: 0;
  padding: 0.5em;
  color: #fff;
  &:focus {
    outline: none;
    box-shadow: 0px 0px 0px;
  }
`;
