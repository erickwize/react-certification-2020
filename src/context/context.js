// import Dexie from 'dexie'
import React, { useEffect, createContext, useReducer, memo } from 'react'
import { useFetch } from '../utils/hooks/useFetch'

export const Context = createContext(null)

export const ContextReducer = (state, action) => {
  switch (action.type) {
    case 'TOGGLE_THEME':
      return {
        ...state,
        theme: !state.theme,
      }
    case 'OPEN_MENU':
      return {
        ...state,
        menu: action.payload,
      }
    case 'GET_DATA':
      return {
        ...state,
        data: action.payload,
      }
    case 'SET_VIEW':
      return {
        ...state,
        view: action.payload,
      }
    case 'SET_TARGET':
      return {
        ...state,
        target: action.payload,
      }
    default:
      return state
  }
}

export const initialState = {
  target: 'Wizeline',
  theme: false,
  menu: false,
  view: 'home',
  data: [],
}
export const ContextProvider = ({ children }) => {
  const [state, dispatch] = useReducer(ContextReducer, initialState)
  const { data, isLoading } = useFetch(state.target)

  useEffect(() => {
    if (data) dispatch({ type: 'GET_DATA', payload: data })
  }, [data])

  return isLoading ? (
    <div> Loading...</div>
  ) : (
    <Context.Provider value={{ state, dispatch }}> {children} </Context.Provider>
  )
}
export default memo(ContextProvider)
