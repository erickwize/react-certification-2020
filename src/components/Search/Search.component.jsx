import React, {useState} from 'react';
import useInput from './../../utils/hooks/useInput';
import { Input, InputContainer, Icon } from './Search.styles';

function Search() {

  const [value, handleInputChange] = useInput('wizeline');

  const submit = e => {
    e.preventDefault();
   console.log(value)
  }

  return(
    <InputContainer onSubmit={submit}>
    <Icon />
      <Input value={value} onChange={handleInputChange} type="text" />
    </InputContainer>
  )
};

export default Search;
