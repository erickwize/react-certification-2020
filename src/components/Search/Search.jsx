import React from 'react';
import { useHistory } from 'react-router-dom';

import { SearchDiv, InputElement } from './Search.styled';

const Search = ({ handleChange }) => {
  const history = useHistory();

  const onHandleSearch = (event) => {
    if (event.key === 'Enter') {
      handleChange(event.target.value);
      history.push('/');
    }
  };

  return (
    <SearchDiv>
      <InputElement
        className="inputSearch"
        type="search"
        placeholder="Buscar contenido"
        onKeyDown={onHandleSearch}
      />
    </SearchDiv>
  );
};

export default Search;
