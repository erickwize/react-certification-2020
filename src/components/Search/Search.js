import React, { useCallback, useContext } from 'react'
import Input from '../Input/Input'
import StyledSearch from './Search.styled'
import { Context } from '../../context/context'

const Search = React.memo(() => {
  const { dispatch } = useContext(Context)
  const handleChange = useCallback(
    (e) => {
      if (e.key === 'Enter') {
        dispatch({ type: 'SET_TARGET', payload: e.target.value })
        dispatch({ type: 'OPEN_MENU', payload: false })
        dispatch({ type: 'SET_VIEW', payload: 'home' })
      }
    },
    [dispatch]
  )
  return (
    <StyledSearch>
      <Input customClickEvent={handleChange} />
    </StyledSearch>
  )
})

export default Search
