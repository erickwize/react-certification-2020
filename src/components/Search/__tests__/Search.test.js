import React from 'react'
import renderer from 'react-test-renderer'
import { Context } from '../../../context/context'
import Search from '../Search'

const state = {
  target: 'Wizeline',
  theme: false,
  menu: false,
  view: 'home',
}
describe('Testing Search component', () => {
  test('Renders Search without crashing', async () => {
    const rendered = renderer.create(
      <Context.Provider value={{ state }}>
        <Search />
      </Context.Provider>
    )
    expect(rendered).toMatchSnapshot()
  })
})
