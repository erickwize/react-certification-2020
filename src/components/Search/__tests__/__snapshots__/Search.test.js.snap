// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Search component Renders Search without crashing 1`] = `
<div
  className="sc-gsDJrp dLVeSJ"
>
  <input
    className="sc-bdvvaa gzIPOt"
    onKeyDown={[Function]}
    placeholder="Wizeline"
    role="input"
    type="text"
  />
</div>
`;
